### ISSUE TITLE
[bug] CONAN_CMAKE_GENERATOR_PLATFORM is not respected in CMakeToolchain generate

### ISSUE THREAD

**Issue by furhat** [conan-io/conan#13825]
We have a use case to do the cross build for Windows ARM64EC mode, but no way to make it work if I use the CMakeToolchain generator.

### Environment details

* Operating System+version:Windows
* Compiler+version:
* Conan version:1.59.0
* Python version:3.10.10

### Steps to reproduce

1. Set CONAN_CMAKE_GENERATOR_PLATFORM=ARM64EC in my profile
2. in the conan_toolchain.cmake file the CMAKE_GENERATOR_PLATFORM is still set to ARM64

**Comment by furhat** [comment]
In the conan code, the generator platform is derived from the arch, it should first respect the CONAN_CMAKE_GENERATOR_PLATFORM. 
```
 def _get_generator_platform(self, generator):
        settings = self._conanfile.settings
        # Returns the generator platform to be used by CMake
        compiler = settings.get_safe("compiler")
        arch = settings.get_safe("arch")

        if settings.get_safe("os") == "WindowsCE":
            return settings.get_safe("os.platform")

        if compiler in ("msvc", "clang") and generator and "Visual" in generator:
            return {"x86": "Win32",
                    "x86_64": "x64",
                    "armv7": "ARM",
                    "armv8": "ARM64"}.get(arch)
        return None
```

**Comment by memsharded** [comment]
Hi @furhat 

This is not a bug, but expected behavior.
New generators like ``CMakeToolchain`` have dropped support for env-vars like ``CONAN_CMAKE...`` and receive their configuration via ``conf``. Please have a look to this page: https://docs.conan.io/1/reference/conanfile/tools/cmake/cmaketoolchain.html

**Comment by furhat** [comment]
Hi @memsharded , thanks for the comment.
I know Conan provides the capability for [Extending and customizing CMakeToolchain](https://docs.conan.io/1/reference/conanfile/tools/cmake/cmaketoolchain.html#extending-and-customizing-cmaketoolchain), but looks like it is based on that I own the conanfile. In my case, I would like to affect some Conan packages that my package depends on. It seems I have no chance to do the customization like 
```
    tmp = tc.blocks["generic_system"].template
    new_tmp = tmp.replace(...)  # replace, fully replace, append...
    tc.blocks["generic_system"].template = new_tmp
```
It still seems to me I have no chance to change the `CMAKE_GENERATOR_PLATFORM` for my upstream package.

I would like to suggest adding a conf variable like `tools.cmake.cmaketoolchain:generator_platform` to control the `-A` parameter used by the cmake CLI.

**Comment by memsharded** [comment]
Quick question, is there any other mapping, input, setting that could be leveraged to define ``ARM64EC``? Shouldn't that architecture be reflected in settings too, if it creates a different binary? What are your input settings?

If it makes sense we could extend the current ``CMAKE_SYSTEM_NAME`` et al:
```
system_name = self._conanfile.conf.get("tools.cmake.cmaketoolchain:system_name")
system_version = self._conanfile.conf.get("tools.cmake.cmaketoolchain:system_version")
system_processor = self._conanfile.conf.get("tools.cmake.cmaketoolchain:system_processor")
```
and add the ``generator_platform``, but I'd like to understand better the use case first and check if there is no other input that could be used.

**Comment by furhat** [comment]
By now, `CONAN_CMAKE_GENERATOR_PLATFORM` environment variable (yields `CMAKE_GENERATOR_PLATFORM`) is the only way I could find to make it work. But as I reported, this only works for my conan package using cmake generator but for the upstream packages, this env
I agree it make more sense to have it reflected in settings (probably `settings.arch`), then eventually be reflected in the `cmake -A`  or `CMAKE_GENERATOR_PLATFORM`. 
You can find more information for ARM64EC from [Get started with Arm64EC](https://learn.microsoft.com/en-us/windows/arm/arm64ec-build).

**Comment by memsharded** [comment]
I'd say that this is a new Windows-specific new architecture, and as such, it could deserve be in the default ``settings.yml``.

Any suggestion @jcar87 ?

**Comment by memsharded** [comment]
Hi @furhat 

https://github.com/conan-io/conan/pull/14114 has been merged for 2.0.7 which means:

- The ``arm64ec`` will be in the default settings.yml
- The CMAKE_GENERATOR_PLATFORM will be set to ARM64EC for this architecture and ``Visual Studio`` generators

However, note that the support for this architecture is still very preliminar even for CMake: https://gitlab.kitware.com/cmake/cmake/-/issues/21724, and there might be other things missing. So the support for this new architecture is considered very **experimental** and will probably require time and some iterations. Having more feedback here will be very important, so looking forward your feedback. Many thanks!

**Comment by furhat** [comment]
@memsharded , we will evaluate it and let you know the result.
Many thanks for the quick response!
